### DISCUSSION TITLE
server.executeOperation creating context without passing data in

### DISCUSSION THREAD

**Discussion by Ustice** [apollographql/apollo-server#5008] [Q&A]
I have a project that is using `apollo-server-express` and supporting both a standard graphql endpoint, and several other REST endpoints. I want to be able to run queries and mutations from elsewhere in my code (including from those REST endpoints that I mentioned). The best way that I have found do that is to use the `server.executeOperation` method, which does what I want; however it generates a context as a part of my standard context generator function _without passing my `express` `Request` or `Response` objects to the context generator.

This behavior is described in [#2886](https://github.com/apollographql/apollo-server/issues/2886).

I see that there is a [PR which adds the ability](https://github.com/apollographql/apollo-server/pull/4166/commits/588d623306200aa5830ce71cd297fcbdded13623) to use and alternative context object, but it was never merged. 

I have two questions:

  1. Should I be using some other function to programmatically query the ApolloServer?
  2. Was the PR listed above never merged because the author didn't sign the CLA, are there any obstacles to me submitting an updated PR to resolve this issue (assuming that I sign the CLA)?

**Comment by glasser** [comment]
I've just fixed #2886 (which should go out in a future version, probably 2.23.0).
